tests/python/unittest/test_target_codegen_hexagon.py

```diff
@@ -17,30 +17,21 @@
 
 import numpy as np
 import os
+import pytest
 import re
+import sys
 import tvm
 import tvm.relay
+import tvm.testing
 import tvm.contrib.hexagon as hexagon
 
 
-def check_prereq_and_setup():
-    if tvm.target.codegen.llvm_version_major() <= 7:
-        print("Skipping test: need LLVM 7 or later for codegen")
-        return False
-    if os.name != "posix":
-        print("Skipping test on non-POSIX platforms")
-        return False
-    if not tvm.runtime.enabled("hexagon"):
-        print("Hexagon runtime not enabled")
-        return False
-    # Register a phony linker, so that we can test codegen without a Hexagon toolchain.
-    hexagon.register_linker(lambda: "/bin/true")
-    return True
+# Register a phony linker, so that we can test codegen without a Hexagon toolchain.
+hexagon.register_linker(lambda: "/bin/true")
```

Comment:
should this become a fixture and unregister on test exit?

Reply:
Yeah, that would be cleaner.

Reply:
One problem is that if all of these tests run from a common "super"-module, and are executed in parallel, then the linker may get unregistered by one test before the other tests are done using it.  Right now this is the only tests that does this, but if there are more added later, then it may become an issue.

Another thing may be that we may want to do lazy linking in Hexagon module, so that we're not looking for the linker each time when `tvm.build` is executed...

Reply:
pytest-xdist will spawn new sys.executable to distribute the load, so shouldn't need to worry about refcounting here. the fixture should be enough. i am going to push on the xdist PR again soon, which is why i ask :). do you mind making the change here? it's not too complicated--just `@pytest.fixture` around a local function, move the register call there, and take the function name as an argument to each test_

Reply:
I've turned it into a fixture.